watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1163 @@
+        if temperature is not None:
+            if float(temperature):
+                temp_input.update({"value": float(temperature)})
+            else:
+                raise RuntimeError(
+                    f"Invalid temperature: {temperature}. Expected number"
+                )
+        input_list.append(temp_input)
 
-        _name = lambda k: get_oli_name(k) if get_oli_names else k
+        pres_input = {
+            "group": "Properties",
+            "name": "Pressure",
+            "unit": self.input_unit_set["pressure"]["oli_unit"],
+            "value": 101325,
+        }
+        if pressure is not None:
+            if float(pressure):
+                pres_input.update({"value": float(pressure)})
+            else:
+                raise RuntimeError(f"Invalid pressure: {pressure}. Expected number")
+        input_list.append(pres_input)
+
+        reconciliation_options = [
+            "EquilCalcOnly",
+            "ReconcilePh",
+            "ReconcilePhAndAlkalinity",
+            "ReconcilePhAndAlkalinityAndTic",
+        ]
+        rec_input = {
+            "group": "Calculation Options",
+            "name": "CalcType",
+            "value": "EquilCalcOnly",
+        }
+        if reconciliation is not None:
+            if reconciliation in reconciliation_options:
+                rec_input.update({"value": reconciliation})
+            else:
+                raise RuntimeError(
+                    f"Invalid reconciliation option: {reconciliation}."
+                    + f" Use one of {reconciliation_options}"
+                )
+        else:
+            reconciliation = "EquilCalcOnly"
+
+        input_list.append(rec_input)
+        additional_req_input = []
+        additional_req_args = []
+        if "Ph" in reconciliation:
+            additional_req_args.append([ph, acid_titrant, base_titrant])
+            if not acid_titrant:
+                acid_titrant = "HCl"
+            if not base_titrant:
+                base_titrant = "NaOH"
+            additional_req_input.extend(
+                [
+                    {
+                        "group": "Properties",
+                        "name": "pH",
+                        "value": ph,
+                    },
+                    {
+                        "group": "Calculation Options",
+                        "name": "PhAcidTitrant",
+                        "value": get_oli_name(acid_titrant),
+                    },
+                    {
+                        "group": "Calculation Options",
+                        "name": "PhBaseTitrant",
+                        "value": get_oli_name(base_titrant),
+                    },
+                ]
+            )
+        if "Alk" in reconciliation:
+            additional_req_args.append([alkalinity, alkalinity_ph, alkalinity_titrant])
+            if not alkalinity_titrant:
+                alkalinity_titrant = "H2SO4"
+            if not alkalinity_ph:
+                alkalinity_ph = 4.5
```

Comment:
Any reason why this is hard-coded to 4.5?

Reply:
I am guessing this is just a ballpark default if the user doesn't provide any value for this. Should we let the user know that this is the case with logger output?